components/raftstore/src/store/config.rs

```diff
@@ -161,6 +161,11 @@ pub struct Config {
     pub hibernate_regions: bool,
     #[config(hidden)]
     pub early_apply: bool,
+    // Only used for test
+    #[doc(hidden)]
+    #[serde(skip_serializing)]
+    #[config(skip)]
+    pub merge_ensure_all_target_peer_exist: bool,
```

Comment:
I suggest to only keep one configuration and change this to `dev_assert` instead. It's redundant to keep multiple configuration entries to just enable assert for tests.

Reply:
+1

Reply:
@Connor1996 🤣 sorry for deleting your +1 by accident

`merge_ensure_all_target_peer_exist` is just used for passing some old tests. In other words, it can be removed if these old tests have removed.
So it has a different purpose with `dev_assert`.

Reply:
You can also make old tests pass by setting `dev_assert` to false. It's just a redundant configuration.

Reply:
`dev_assert` config is a global variable so we cannot set it just for some tests.
Besides, `dev_assert` can be used in many places not only in this merge case. For example, these old tests may panic in other `dev_assert` due to some new PRs. Just turn it down may miss some errors.

Reply:
I don't think the case deserve such complexity. Otherwise, we may need a separate configurations for every corner cases. I have already suggested to change the name to dev_assert and remove the global one. That way we can control it more precisely.

For future potential usage, I think we can wait for tikv/fail-rs#55, which can solve the case easily.